### PR TITLE
circle-line-collision

### PR DESCRIPTION
**Pull request by joydeep-b**

### CODE REVIEW

**Comment by kylevedder** [comment]
For `CheckLineLineCollision()` we consider touching to be a collision (https://github.com/ut-amrl/amrl_shared_lib/blob/master/math/geometry.h#L155) and thus I think we should consider the line touching the circle to be a collision.

As such, I suggest changing all the `< r` to `<= r`.

File: math/geometry.h [hunk 1]
```diff
@@ -494,6 +494,19 @@ T MinDistanceLineLine(const Eigen::Matrix<T, 2, 1>& a0,
...
+// Check if the circle with center c0, and radius r collides with the line 
+// segment p0 : p1, and return true if a collision is found.
+template <typename T>
+bool CheckCircleLineCollision(const Eigen::Matrix<T, 2, 1>& c0,
+                              const T& r,
+                              const Eigen::Matrix<T, 2, 1>& p0,
+                              const Eigen::Matrix<T, 2, 1>& p1) {
+  if ((c0 - p0).squaredNorm() < math_util::Sq(r)) return true;
+  if ((c0 - p1).squaredNorm() < math_util::Sq(r)) return true;
+  if (DistanceFromLineSegment(c0, p0, p1) < r) return true;
```
Comment (kylevedder) [inline math/geometry.h]: Convert `< r` to `<= r` to be consistent with other collision functions.

**Comment by joydeep-b** [comment]
Good catch, updated and added tests for that.

**Comment by kylevedder** [comment]
LGTM

**Comment by joydeep-b** [comment]
Thanks!

**Pull request merged**